runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -327,6 +327,108 @@ void J9::RecognizedCallTransformer::process_java_lang_StringUTF16_toBytes(TR::Tr
    treetop->insertAfter(TR::TreeTop::create(comp(), TR::Node::create(node, TR::treetop, 1, newCallNode)));
    }
 
+/*
+This method inlines a call to ArraysSupport.vectorizedMismatch to a node equivalent to the following pseudocode
+
+   lengthInBytes = length << log2ArrayIndexScale
+   mask = (log2ArrayIndexScale<<1) | 3
+   n = lengthInBytes & ~(mask)
+   res = arrayCmpLen(a+aOffset, b+bOffset, n)
+   if (res == n)  // no mismatch found
+      return ~((lengthInBytes & mask) >> log2ArrayIndexScale)
+   else           // mismatch found
+      return res >> log2ArrayIndexScale
+
+Node before the transformation:
+
+   icall  jdk/internal/util/ArraysSupport.vectorizedMismatch(Ljava/lang/Object;JLjava/lang/Object;JII)I
+      <a>
+      <aOffset>
+      <b>
+      <bOffset>
+      <length>
+      <log2ArrayIndexScale>
+
+Node after the transformation:
+
+   iselect ()
+      icmpeq
+         arraycmp (arrayCmpLen )
+            aladd
+               <a>
+               <aOffset>
+            aladd
+               <b>
+               <bOffset>
+            i2l
+               iand
+                  ishl
+                     <length>
+                     <log2ArrayIndexScale>
+                  ixor
+                     ior
+                        ishl
+                           <log2ArrayIndexScale>
+                           iconst 1
+                        iconst 3
+                     iconst -1
+         ==>iand
+      ixor
+         ishr
+            iand
+               ==>ishl
+               ==>ior
+            <log2ArrayIndexScale>
+         ==>iconst -1
+      ishr
+         ==>arraycmp
+         <log2ArrayIndexScale>
+*/
+void J9::RecognizedCallTransformer::process_jdk_internal_util_ArraysSupport_vectorizedMismatch(TR::TreeTop* treetop, TR::Node* node)
+   {
+   TR::Node* a = node->getChild(0);
+   TR::Node* aOffset = node->getChild(1);
+   TR::Node* b = node->getChild(2);
+   TR::Node* bOffset = node->getChild(3);
+   TR::Node* length = node->getChild(4);
+   TR::Node* log2ArrayIndexScale = node->getChild(5);
+
+   TR::Node* lengthInBytes = TR::Node::create(TR::ishl, 2, length, log2ArrayIndexScale);
```

Comment:
Also looking at the created node, it seems like you may be losing bytecode information from the original call node. Take a look at how we generate trees for other recognized method for reference.